File: nutrinova-api/NutrinovaData/Entities/Patient.cs
using System;
using System.Collections.Generic;

namespace NutrinovaData.Entities;

public partial class Patient
{
  public Guid Id { get; set; }

  public string Firstname { get; set; } = null!;

  public string? Lastname { get; set; }

  public short Age { get; set; }

  public string? ProfilePictureName { get; set; }

  public string? Sex { get; set; }

  public Guid? CustomerId { get; set; }

  public bool Optedoutpatientdetails { get; set; }

  public virtual Customer? Customer { get; set; }

  public virtual ICollection<Meal> Meals { get; set; } = new List<Meal>();

  public virtual ICollection<PatientNutrientDailyGoal> PatientNutrientDailyGoals { get; set; } = new List<PatientNutrientDailyGoal>();
}

File: nutrinova-ui/src/app/(authorized)/patients/_models/patient.ts
export interface Patient {
  id?: string;
  firstname: string;
  lastname?: string;
  customerId?: string;
  age: number;
  sex?: 'M' | 'F' | 'O';
  base64image?: string;
  hasPicture: boolean;
}

export interface CreatePatientReq extends Patient {
  useDefaultNutrientGoals: boolean;
}